### PR TITLE
Fbttf 208 hud

### PR DESCRIPTION
**Pull request by romainpanno**

### CODE REVIEW

File: src/hud/manage_hud.c [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        return RET_ERR_INPUT;
+    }
+    if (add_hud_to_uid_element(scene, head_with_borders, player) != RET_OK) {
+        return RET_ERR_INPUT;
+    }
+    return RET_OK;
+}
+
+int init_hud(window_t *win, scene_t *scene)
+{
```
Comment (Saverio976) [inline src/hud/manage_hud.c:93]: Check scene and win null

File: src/player/update/update_attack.c [hunk 2]
```diff
@@ -9,6 +9,14 @@
 #include "my_rpg.h"
 #include "my_json.h"
 
+static void consume_energy(player_t *player)
+{
+    if (player->energy < 0) {
```
Comment (Saverio976) [inline src/player/update/update_attack.c:14]: <= 0

File: src/player/update/update_player.c [hunk 3]
```diff
@@ -24,4 +24,7 @@ void update_player(__attribute__((unused)) object_t *obj, scene_t *scene,
     } else if (player->state >= 0 && player->state <= 2) {
         update_ptr[player->state](player, scene, win, dtime);
     }
+    if (player->life < 0 && player->state != DIE) {
```
Comment (Saverio976) [inline src/player/update/update_player.c:27]: <= 0

**Review by Saverio976 — CHANGES_REQUESTED** [review]

**Comment by Saverio976** [comment]
Did this will kill the link player if we move ?

File: include/my_rpg.h [hunk 4]
```diff
@@ -55,6 +55,8 @@ typedef struct player_s {
     dir_t dir;
     object_t *obj;
     sfView *view;
+    float life;
```
[the remark this answers is not in the document]
Comment (romainpanno) [reply include/my_rpg.h:58]: In the json it is the max of the stat not the actual value of the stat

[the remark this answers is not in the document]
Comment (Saverio976) [reply include/my_rpg.h:58]: Im dumb

[the remark this answers is not in the document]
Comment (romainpanno) [reply include/my_rpg.h:58]: Me too don't worry

**Review by Saverio976 — APPROVED** [review]

**Pull request merged**